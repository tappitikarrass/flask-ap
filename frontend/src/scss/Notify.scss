@use "base16" as *;

.notify-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: left;

    z-index: 1;
    position: fixed;
    top: 0;
    right: 0;

    width: 30vw;

    margin: 0.5em;
    margin-top: 10vh;
    padding: 1em;

    box-shadow: 0px 0px 20px black;
    border-radius: 5px;

    opacity: 0;
    transition: opacity 1s;
}

.notify-error {
    @extend .notify-container;
    background-color: $base08;
}

.notify-warn{
    @extend .notify-container;
    background-color: $base09;
}

.notify-msg {
    @extend .notify-container;
    background-color: $base0D;
}

.notify-title,
.notify-description {
    margin: 0;
    padding: 0;
}

.notify-title {
    margin-bottom: 0.5em;
}


.notify-text {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

// mediaqueries
// desktop
@media screen and (min-width: 810px) {
}

// tablet
@media screen and (min-width: 500px) and (max-width: 810px) {
    .notify-container {
        width: 50vw;
    }
}

// phone
@media screen and (max-width: 500px) {
    .notify-container {
        width: 90vw;
        margin-left: 5vw;
        margin-right: 5vw;
    }
}
