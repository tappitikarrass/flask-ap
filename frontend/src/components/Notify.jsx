import React from 'react';
import PropTypes from 'prop-types';
import '../scss/Notify.scss';

function Notify({
  id, type, title, msg,
}) {
  return (
    <div id={id} className={`notify-${type}`}>
      <div className="notify-text">
        <h3 className="notify-title">{title}</h3>
        <p className="notify-description">
          {msg}
        </p>
      </div>
    </div>
  );
}

const NotifyType = {
  msg: 'msg',
  warn: 'warn',
  error: 'error',
};

Notify.propTypes = {
  type: PropTypes.oneOf(Object.keys(NotifyType)),
  title: PropTypes.string,
  id: PropTypes.string,
  msg: PropTypes.string,
};

Notify.defaultProps = {
  type: 'msg',
  title: 'Title',
  msg: 'message',
  id: 'notification',
};

export default Notify;
