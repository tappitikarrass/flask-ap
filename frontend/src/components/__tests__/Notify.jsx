import React from "react";
import { HashRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import Notify from '../Notify';

describe('Notify', () => {
  it('',() => {
    let component;
    act(() => {
      component = create(
        <HashRouter>
          <Notify />
        </HashRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
