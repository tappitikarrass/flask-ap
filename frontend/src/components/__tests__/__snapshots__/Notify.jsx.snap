// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notify  1`] = `
<div
  className="notify-msg"
  id="notification"
>
  <div
    className="notify-text"
  >
    <h3
      className="notify-title"
    >
      Title
    </h3>
    <p
      className="notify-description"
    >
      message
    </p>
  </div>
</div>
`;
